boofuzz/ssl_socket_connection.py

```diff
@@ -0,0 +1,89 @@
+from __future__ import absolute_import
+import ssl
+
+from future.utils import raise_
+from . import exception, tcp_socket_connection
+
+
+class SSLSocketConnection(tcp_socket_connection.TCPSocketConnection):
+    """BaseSocketConnection implementation for use with SSL Sockets.
+
+    Args:
+        host (str): Hostname or IP adress of target system.
+        port (int): Port of target service.
+        send_timeout (float): Seconds to wait for send before timing out. Default 5.0.
+        recv_timeout (float): Seconds to wait for recv before timing out. Default 5.0.
+        server (bool): Set to True to enable server side fuzzing.
+        sslcontext (ssl.SSLContext): Python SSL context to be used. Required if server=True or server_hostname=None.
+        server_hostname (string): server_hostname, required for verifying identity of remote SSL/TLS server
+    """
+
+    def __init__(
+        self, host, port, send_timeout=5.0, recv_timeout=5.0, server=False, sslcontext=None, server_hostname=None
+    ):
+        super(SSLSocketConnection, self).__init__(host, port, send_timeout, recv_timeout, server)
+
+        self.sslcontext = sslcontext
+        self.server_hostname = server_hostname
+
+        if self.server is True and self.sslcontext is None:
+            raise ValueError("Parameter sslcontext is required when server=True.")
+        if self.sslcontext is None and self.server_hostname is None:
+            raise ValueError("SSL/TLS requires either sslcontext or server_hostname to be set.")
+
+    def open(self):
+        super(SSLSocketConnection, self).open()
+
+        # If boofuzz is the SSL client and user did not give us a SSLContext,
+        # then we just use a default one.
+        if self.server is False and self.sslcontext is None:
+            self.sslcontext = ssl.create_default_context(ssl.Purpose.SERVER_AUTH)
+            self.sslcontext.check_hostname = True
+            self.sslcontext.verify_flags = ssl.CERT_REQUIRED
```

Comment:
```suggestion
            self.sslcontext.verify_mode = ssl.CERT_REQUIRED
```
Took me some time debugging this :)